jena-arq/src/main/java/org/apache/jena/riot/system/Checker.java

```diff
@@ -206,36 +217,45 @@ public static boolean checkLiteral(String lexicalForm, String lang, ErrorHandler
 
     public static boolean checkLiteral(String lexicalForm, String lang, RDFDatatype datatype, ErrorHandler errorHandler, long line,
                                        long col) {
-        boolean hasLang = lang != null && !lang.equals("");
-        if ( !hasLang ) {
-            // Datatype check (and RDF 1.0 simple literals are always well formed)
-            if ( datatype != null )
-                return validateByDatatype(lexicalForm, datatype, errorHandler, line, col);
-            return true;
-        }
-
-        // Has a language.
-        if ( JenaRuntime.isRDF11 ) {
-            if ( datatype != null && !Objects.equals(datatype.getURI(), NodeConst.rdfLangString.getURI()) ) {
-                errorHandler(errorHandler).error("Literal has language but wrong datatype", line, col);
-                return false;
+        boolean hasLang = ( lang != null && !lang.isEmpty() );
+        boolean hasDatatype = datatype != null;
+
+        // Has language...
+        if ( hasLang ) {
+            // Has datatype (generally, a problem)...
+            if ( hasDatatype) {
+                // Jena is using RDF 1.1...
+                if ( JenaRuntime.isRDF11 ) {
+                    // The datatype is NOT "rdf:langString"...
+                    if ( datatype.getURI() != NodeConst.rdfLangString.getURI() ) {
```

Comment:
Use `Objects.equals` here. Strings should not be compared with `=` or `!=` which is "same object" not "same string".

Reply:
Indeed!  I'll update to: ( ! datatype.getURI().equals( NodeConst.rdfLangString.getURI() )